cmake/api_macros.cmake

```diff
@@ -191,6 +191,21 @@ macro(OPENDDS_TARGET_SOURCES target)
     list(APPEND _tao_options "-SS")
   endif()
 
+  list(LENGTH ${CMAKE_CXX_COMPILER} cxx_compiler_length)
+
+  if(${cxx_compiler_length} LESS 1)
+    if(NOT "${_opendds_options}" MATCHES "-Yp")
+      list(APPEND _opendds_options "-Yp,${CMAKE_CXX_COMPILER}")
+    endif()
+
+    if(NOT "${_tao_options}" MATCHES "-Yp")
+      list(APPEND _tao_options "-Yp,${CMAKE_CXX_COMPILER}")
+    endif()
+
+  else()
+    message(FATAL_ERROR "Unable to use the default CMake CXX compiler, please add it to the path.")
```

Comment:
I don't think this is a concern as long as the project is a C++ project. I would change this to say we don't support argument items in `CMAKE_CXX_COMPILER`.